<?xml version="1.0"?>
<robot name="lab">

  <material name="silver">
    <color rgba="0.7 0.7 0.7 1"/>
  </material>

  <material name="grey">
    <color rgba="0.9 0.9 0.9 1"/>
  </material>

  <link name="beam">
    <visual>
      <geometry>
        <box size="3 0.1 0.1"/>
      </geometry>
      <material name="grey"/>
      <origin xyz="1 0 0.5"/>
    </visual>
  </link>

  <joint name="base_to_pipe_x" type="continuous">
    <parent link="beam"/>
    <child link="steel_pipe_x_dummy"/>
    <axis xyz="1 0 0"/>
    <origin xyz="1 0 0.5"/>
  </joint>

  <link name="steel_pipe_x_dummy" />

  <joint name="base_to_pipe_y" type="continuous">
    <parent link="steel_pipe_x_dummy"/>
    <child link="steel_pipe_y_dummy"/>
    <axis xyz="0 1 0"/>
    <origin xyz="0 0 0"/>
  </joint>

  <link name="steel_pipe_y_dummy" />

  <joint name="base_to_pipe_z" type="continuous">
    <parent link="steel_pipe_y_dummy"/>
    <child link="steel_pipe"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0 0 0"/>
  </joint>

  <link name="steel_pipe">
    <visual>
      <geometry>
        <cylinder length="0.7" radius="0.05"/>
      </geometry>
      <origin xyz="0 0 -0.5"/>
      <material name="silver"/>
    </visual>
  </link>

</robot>

